### DISCUSSION TITLE
Zipped files and provider-reference files

### DISCUSSION THREAD

**Discussion by olderthandirt5** [CMSgov/price-transparency-guide#473] [Q&A] [answered]
@shaselton-usds 
If we use an  in-network file with separate provider-reference files, can these be zipped together into one file for download?

If that is permissible, would we still need to have each separate provider-reference file available for download?

So to be clear, are any of these allowable?:

1. One zipped file containing one in-network file and multiple (25-325) provider-reference files
2. One zipped in-network file and one zipped provider-reference file containing all the individual files
3. If either option 1 or 2 is permissible, would we also need to make the individual provider-reference files available for download individually?

What sparked this discussion was the requirement to specify a FQDN for the external provider-reference files and the references I could find to zipped files predated the use of external provider-reference files.

**Comment by olderthandirt5** [comment]
@shaselton-usds 
Any guidance on this? We are hoping to publish files for one of our partners on Monday.

Thanks.

**Reply by Data-san** [reply]
@olderthandirt5 Did you publish on Monday? If so, what did you decide? Helpful perspective. Thanks.

**Reply by Devyandu** [reply]
The in-network file needs to have a provider-reference object which has the  location of the provider reference.
For either 1 or 2 what would you put in the location if the provider reference file is not even hosted on it's own anywhere?

FYI - We are zipping all the provider reference files individually and then providing the full URL of each provider reference file inside the "provider_reference.location" attribute of the in-network file

**Reply by Data-san** [reply]
@Devyandu How many provider reference files did you have to zip? Thanks.

**Reply by Devyandu** [reply]
Approximately 500-1000 provider reference files per each in-network file. 
Most provider reference files are fairly small.
Many provider reference files will shared amongst multiple  in-network files.

**Comment by olderthandirt5** [comment]
@Data-san Without any guidance from CMS yet, we published a zip file that had all of files (TOC, in-network-rates, and 80+ provider-reference files), and each individual file separately. This set will be consumed and republished by a partner, so the files will be modified before posting to the public.

When we publish our self-hosted files I hope we will have some sort of rules to follow.

@Devyandu We put the location of the individual provider-reference files. At this point, the zip file is a convenience for users.

**Comment by shaselton-usds** [comment, marked as answer]
@olderthandirt5 - I don't think option 1 makes much sense zipping everything together; at that point, why not just put everything in a single file?

Option 2 makes a bit more sense, but you are losing the ability for sharing or reusing individual provider networks across multiple in-network files if all your individual files are grouped together, which was one of the reasons for allowing that ability -- meaning, you could still do [inline provider referencing](https://github.com/CMSgov/price-transparency-guide/blob/59a3adc4f9d1ef6f86a306ea8714112db1805185/examples/in-network-rates/in-network-rates-fee-for-service-single-plan-sample.json#L10-L25) for the in-network file.

To take advantage of the external provider referencing ability, I'd suggest @Devyandu's approach with zipping each file up individually and then referencing them in the appropriate in-network files. Also, @Devyandu is correct, those provider reference files need a public location to be accessed, otherwise what is going in the `location` attribute?
